### ISSUE TITLE
Feature Request - Please support ActivityPub for updates to threads

### ISSUE THREAD

**Issue by DerekCaelin** [loomio/loomio#9527]
My co-op uses Loomio for governance and a Mastodon server for communication. It would be great if we could see the results of polls, updates on discussions, etc, showing up our feeds. Supporting ActivityPub in the way that [Discourse is now doing](https://meta.discourse.org/t/federation-support-for-discourse/90921?page=5) would be very valuable for our community.

**Comment by robguthrie** [comment]
Hi Derek, Could you explain how the feature would be useful to you?

Do you know anyone interested in doing the development work?

**Comment by poVoq** [comment]
It would be similar to an RSS feed, but possible to follow from Mastodon etc. Discourse is indeed working on such limited ActivityPub support.

For now as a work-around I would suggest you just use a RSS feed to Mastodon bot that should already work with Loomio.

**Comment by almereyda** [comment]
Indeed, with regards to [pace-layering](https://jods.mitpress.mit.edu/pub/issue3-brand/release/2) an organisations activity, it is useful for groups to bridge conversations across platforms.

An ActivityPub integration of Loomio would render it an equal participant in conversations that already today span multiple channels, as we see with the aforementioned Mastodon and Discourse, but also with WordPress, Pixelfed, Peertube, Owncast, WriteFreely and the likes. Offering an integration with ActivityPub allows to collect various dispersed conversation streams into more linear narratives and helps to structure web-wide conversation.

Currently, I'm seeing these stories present in this thread:

> **As a** participant in a Loomio discussion,
>
> **I need to** be informed about new interactions on a secondary ActivityPub channel,
>
> **in order to** be able to react to the centrally held conversation from a peer actor on a federated platform and to keep an additional copy, say audit log of the conversation.

> **As a** participant in a Loomio discussion,
>
> **I need to** respond to new interactions via a secondary ActivityPub channel,
>
> **in order to** participate in the centrally held conversation from a peer actor on a federated platform that keeps an additional copy of the conversation.

If you are unsure what else ActivityPub support could be good for, please find inspiration in the replies to this recent toot: https://cosocial.ca/@evan/112185826897209052

**Comment by Hamsterarsch** [comment]
Hi everyone, might look into this at some point but I think it would be good to clarify real-world usecases to make sure development doesn't miss requirements. Clarifications would also be great to guesstimate urgency of the suggested features and feature prioritization. 

Some ideas about implementation without having more clarity:
Seems like the easiest way to get anything on to mastodon and thus fediverse would be to provide a connector to a mastodon account like described [here](https://mcqn.com/posts/simple-posting-to-mastodon-with-ruby/) (see also [mastodon's API](https://docs.joinmastodon.org/client/intro/)). This would require an existing account and dependency on mastodon itself.

If posting to mastodon is not enough for the potential features, it seems the most efficient way right now is to contribute to [this ruby library](https://gitlab.com/experimentslabs/federails). It's still in heavy development but also the only promising framework I found for fediverse support. If you know more, let me know. In case this is the route despite maintenance concerns, this [documentation](https://socialhub.activitypub.rocks/pub/guide-for-new-activitypub-implementers) should give great clarity on the spec.

**Comment by poVoq** [comment]
I would recommend looking into the fediverse integration of NodeBB and Discourse (maybe Lemmy) instead, as those are much closer to Loomio than Mastodon conceptually.

**Comment by Hamsterarsch** [comment]
@poVoq I mentioned mastodon not as a design inspiration but because, as far as I can tell, this request was about posting to mastodon (or, in other words, it's about somehow connecting theads etc. with fediverse):

> My co-op uses Loomio for governance and a Mastodon server for communication. It would be great if we could see the results of polls, updates on discussions, etc, showing up our feeds.

**Comment by robguthrie** [comment]
If someone wants to pick this up, then they should look at the webhooks feature, and convert that into activity pub. 

We're (Loomio Co-op) not going to built this though (someone else can contribute it if they like), so closing the issue.

**Comment by beedell-roke** [comment]
@robguthrie, best to close as unplanned, rather than completed.
